server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -145,8 +145,11 @@ public ClusterState execute(ClusterState currentState) {
                 // validates if correct awareness attributes and forced awareness attribute set to the cluster before starting action
                 validateAwarenessAttribute(decommissionAttribute, awarenessAttributes, forcedAwarenessAttributes);
                 DecommissionAttributeMetadata decommissionAttributeMetadata = currentState.metadata().decommissionAttributeMetadata();
+                if (decommissionAttributeMetadata == null || decommissionAttributeMetadata.status().equals(DecommissionStatus.FAILED)) {
+                    decommissionRequest.originalRequest(true);
```

Comment:
Did you test this? Help me understand during retry how you will the request come with this flag set to true. As it is updating the request object on the master not on the data node which sent the request. The data node will retry with its original request that would have flag set to false.

A simpler option you need is introduce a requestId in DecommissionRequest and generate it in transport class when the new request arrives and store it in the metadata. And you allow reentrant request if it comes with same requestId else fail or if you have status failed in this case you will override the requestId with new requestId at the time of updating the metadata.

Reply:
> Yes, as @Bukhtawar suggested above, will rename it to owningRequest. An owningRequest will be allowed to proceed if a metadata exists with same attribute

Reply:
> Did you test this? Help me understand during retry how you will the request come with this flag set to true. As it is updating the request object on the master not on the data node which sent the request. The data node will retry with its original request that would have flag set to false.

As discussed offline, the retried request won't be routed back to the data node but rather on the same stepped down leader. But anyways, thanks for the suggestion of the request id. Have made the changes to incorporate request id validation to identify if it is a retried request or a new request

Reply:
I think what is needed here is to support versioning for this API. Anytime the metadata is deleted we should still retain the version. This will help cases like below 

INIT(1) || Metadata Delete || INIT(2) at this point two requests can be running concurrently(will fail only when the next state update occurs), which would otherwise have been avoided with versioning.

This would also protect the metadata from being modified by a request with a lower version.

We allow versioning bump for version -1(first request), deletes and failed states

Reply:
Will add a IT for conccurrent Decommission & Recommission action in a follow up PR